simulator/mips/mips_instr.h

```diff
@@ -24,6 +24,38 @@
 #include <array>
 #include <unordered_map>
 
+enum instr_masks {
```

Comment:
That is not an enumeration actually, so I suggest to use `static const constexpr uint32` variables instead

Reply:
Could you please move that stuff to a separate header?